File: T/766-toeplitz-matrix/main.go
package main

import (
	"fmt"
)

func isToeplitzMatrix(matrix [][]int) bool {

	buf := make([]int, len(matrix)+len(matrix[0])-1)
	for i := 0; i < len(matrix); i++ {
		buf[i+len(matrix[0])-1] = matrix[i][0]
	}
	for j := 0; j < len(matrix[0]); j++ {
		buf[-j+len(matrix[0])-1] = matrix[0][j]
	}
	for i := 1; i < len(matrix); i++ {
		for j := 1; j < len(matrix[0]); j++ {
			if buf[i-j+len(matrix[0])-1] != matrix[i][j] {
				return false
			}
		}
	}
	return true
}

func main() {
	fmt.Println("a")
}

File: T/766-toeplitz-matrix/main.rs
struct Solution {}

impl Solution {
  pub fn is_toeplitz_matrix(matrix: Vec<Vec<i32>>) -> bool {
    let row = matrix.len();
    let col = matrix[0].len();
    let mut buf = vec![0; row + col - 1];
    for i in 0..matrix.len() {
      buf[row - 1 - i] = matrix[i][0];
    }
    for i in 1..matrix[0].len() {
      buf[row - 1 + i] = matrix[0][i];
    }
    for i in 0..matrix.len() {
      for j in 0..matrix[0].len() {
        if matrix[i][j] != buf[row - 1 + j - i] {
          return false;
        }
      }
    }
    true
  }
}

fn main() {
  println!("{}", Solution::is_toeplitz_matrix(vec![vec![0; 5]]));
}
